src/Protocol/ForgeHandshake.cpp

```diff
@@ -0,0 +1,365 @@
+
+// ForgeHandshake.cpp
+
+// Implements Forge protocol handshaking
+
+#include "Globals.h"
+#include "ForgeHandshake.h"
+#include "json/json.h"
+#include "../Server.h"
+#include "../Bindings/PluginManager.h"
+#include "../ClientHandle.h"
+#include "../Root.h"
+
+
+/** Discriminator byte values prefixing the FML|HS packets to determine their type. */
+enum
+{
+	Discriminator_ServerHello = 0,
+	Discriminator_ClientHello = 1,
+	Discriminator_ModList = 2,
+	Discriminator_RegistryData = 3,
+	Discriminator_HandshakeReset = -2,
+	Discriminator_HandshakeAck = -1,
+};
```

Comment:
What if you used an enum class? Those have better type safety, also (you can't mix the enum with normal numbers). Also values are scoped within the class (fair warning: I haven't actually used them in a project, I've only just now tried them out in an editor).

https://stackoverflow.com/questions/18335861/why-is-enum-class-preferred-over-plain-enum